typing/typecore.ml

```diff
@@ -6483,6 +6564,11 @@ and type_cases
   else
     (* Check for unused cases, do not delay because of gadts *)
     unused_check false;
+  if may_contain_modules then begin
+    end_def ();
+    (* Ensure that unpacked modules' types do not escape *)
+    unify_exp_types loc env (instance ty_res) (newvar ());
```

Comment:
This certainly looks reasonable, but I'm annoyed we have to do it. I think I want to get the calls to `instance ty_res` elsewhere to use the outer level instead of the ambient one. Let's perhaps talk about this.

Reply:
As discussed, we rolled this in together with the level management for gadts